RIOT-OS/RIOT
# cc26x2_cc13x2: trim device registers on `cpu_init`.

### Contribution description

This is a very stripped-down port from the TI SDK of the
`SetupTrimDevice` function which _must_ be called on CPU initialization.

It sets configurations from the CCFG (Customer Configuration Area), most
of the code is already implemented in the ROM and we use that API.

More importantly, these ROM functions aren't present on `cc26x0` devices
and must be implemented manually on these to just implement
`SetupTrimDevice`, so I opted out on writing a more general
implementation as the code for that is quite big and the review effort
for that kind of code is simply too much (more than writing it IMHO), it
also happens that I don't have one of these to test.

This function is important because it allows us to change the `SCLK_HF`
clock source to the external HF XOSC, without this, the MCU simply halts
when performing the switch due to a bad state on the MCU registers.

Also this function sets some `FLASH` registers to a working state, so we
can write in the future a MTD driver for it.

There's room for improvement in this port, mainly refactorings, but as
documentation is scarse I tried to do a one-to-one port without changing
the logic too much, even some variables don't make sense, or some code
blocks. Unbelievably, it's much worse on the TI SDK, of course we have
the opportunity to change that on RIOT :wink:.

Feel free to criticize the code, as I personally think that can be improved,
but I need some opinions though.

### Testing procedure

- Green travis/murdock.
- Any test/example should run as normal, there isn't a change in the functionality.

### Issues/PRs references

Needed for #13295.
Fixes one item on #13635.
